test/integration_test.py

```diff
@@ -363,11 +366,22 @@ def _prune_test_bundles(self,
                 break
         return filtered_bundle_fqids
 
+    @staticmethod
+    def _filter_superseded_bundle_versions(bundle_fqids: Set[BundleFQID]) -> Set[BundleFQID]:
+        latest_bundle_fqids = set(max(bundle_fqid, key=attrgetter("version"))
```

Comment:
This doesn't quite address my concerns:

https://github.com/DataBiosphere/azul/pull/2042#discussion_r459634915

I would like you to attempt the dict based solution. We can then compare which one is easier to undestand.